lib/autobuild/import/git.rb

```diff
@@ -1278,7 +1316,7 @@ def checkout(package, _options = Hash.new)
                         Autobuild.tool('git'), 'clone', '-o', remote_name, *clone_options,
                         repository, package.importdir, retry: true)
 
-            update_remotes_configuration(package)
+            update_remotes_configuration(package, only_local: false)
             update(package, only_local: !remote_branch.start_with?("refs/"),
```

Comment:
Not super sure about the interaction between shallow and only_local

The update step is needed mainly for setups with a specific commit or tag. With shallow, the commit/tag won't be present locally, and I don't know how that will work out.

Reply:
This is not for shallow, this is for `tag` && `single_branch`. See the other thread.

Reply:
I was commenting on the call to `update`.

Reply:
As far as I understood, `#update` is what sets up the branch so that we are not left with a detached head after cloning with `--single-branch` and `--branch=tag_name`.

Reply:
Ah, I think I get your point now. Good one.

Reply:
I think it's not possible to use shallow and `commit` together. Shallow and `tag` will work if the repository is cloned with `--branch=tag`.

Reply:
In that case, we should probably force shallow to false for repositories that have a specific `commit` set. Since the shallow flag can be set globally, I think that's the best course.

Reply:
Done.